qualtran/bloqs/arithmetic/comparison.py

```diff
@@ -1017,3 +1024,135 @@ def _eq_k() -> EqualsAConstant:
 
 
 _EQUALS_K_DOC = BloqDocSpec(bloq_cls=EqualsAConstant, examples=[_eq_k])
+
+
+@frozen
+class CLinearDepthGreaterThan(Bloq):
+    r"""Controlled greater than between two integers.
+
+    Implements |c>|a>|b>|t> => |a>|b>|t ⨁ ((a > b)c)> using $n+2$ Toffoli gates.
+
+    Note: the true cost is $n+1$ but an extra Toffoli comes from OutOfPlaceAdder which operates
+    on $n+1$ qubits rather than $n$. Changing the definition of OutOfPlaceAdder will remove this
+    extra Toffoli.
+
+    This comparator relies on the fact that (b' + a)' = b - a. If a > b, then b - a < 0. We
+    implement it by flipping all the bits in b, computing the first half of the addition circuit,
+    copying out the carry, and uncomputing the addition circuit.
+
+    Args:
+        dtype: type of the integer registers.
+        cv: ctrl value at which the bloq is active.
+
+    Registers:
+        a: dtype input registers.
+        b: dtype input registers.
+        target: A single bit output register to store the result of a > b.
+
+    References:
+        [Halving the cost of quantum addition](https://arxiv.org/abs/1709.06648).
```

Comment:
nit: typically add more context (section 2, page 4, Eq 7 etc)

Reply:
will do ... I just copied the docstring of LinearDepthGreaterThan